Document ODBC Metadata enumeration (#688)
tableau/connector-plugin-sdk: docs/capabilities.md

CAP_ODBC_UNBIND_EACH | Set to 'yes' to force the Tableau native ODBC protocol to unbind and deallocate columns individually, which may improve stability. | yes | yes 
CAP_ODBC_UNBIND_PARAMETERS_BATCH | Set to ‘yes’ to unbind all parameters in a single batch operation. | yes | yes 

> ### How to read ODBC metadata is determined by capabilities

Reviewer: Can you give this topic its own page, similar to [auth modes](https://tableau.github.io/connector-plugin-sdk/docs/auth-modes): metadata-enumeration.

Basically a general page about what this overall topic is, something like: When connecting to a data source Tableau needs to know hierarchy, and enumerate entities.  When selecting a table or connecting to an existing saved connection the column level info needs to be queried as well.

Then have an ODBC section with this content

Author: ok good idea, will do

1. Full "select *" using SQLPrepare only, which ensures the query is parsed but not executed.
Conditions: CAP_ODBC_METADATA_SUPPRESS_PREPARED_QUERY = false AND CAP_ODBC_METADATA_SUPRESS_SELECT_STAR = false
Example query: `SELECT * FROM default.testv1_batters`

2. Row-limiting "select *" with WHERE 1=0.
Conditions: CAP_QUERY_WHERE_FALSE_METADATA = true AND (CAP_ODBC_METADATA_SUPPRESS_PREPARED_QUERY = false OR CAP_ODBC_METADATA_SUPPRESS_EXECUTED_QUERY = false)
Example query: `SELECT * FROM default.testv1_batters WHERE 1=0`

3. Row-limiting "select *" with TOP 0 / LIMIT 0.
Conditions: CAP_QUERY_TOP_0_METADATA = true AND (CAP_ODBC_METADATA_SUPPRESS_PREPARED_QUERY = false OR CAP_ODBC_METADATA_SUPPRESS_EXECUTED_QUERY = false)
Example query: `SELECT * FROM default.testv1_batters LIMIT 0`

> 4. Less reliable but fast API for querying table metadata directly by SQLGetColumns calls

Reviewer: ```suggestion
4. Less reliable but fast API for querying table metadata directly from [SQLColumns](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolumns-function?view=sql-server-ver15) calls
```

Author: Done, thanks
